proxy/utils.go

```diff
@@ -0,0 +1,29 @@
+package proxy
+
+import (
+	"bytes"
+	"sync"
+)
+
+type LockedBuffer struct {
+	mu  sync.Mutex
+	buf *bytes.Buffer
+}
```

Comment:
I'd use embedding like :)
```go
type SyncBuffer struct {
	bytes.Buffer
	mu  sync.Mutex
}
```

Reply:
you can also embed sync.Mutex, but there are some conditions which are odd, so normally I try to not use embedding